src/rt/sched/mpmcq.h

```diff
@@ -87,9 +88,9 @@ namespace verona::rt
       UNUSED(alloc);
       auto unmasked_node = unmask(node);
       unmasked_node->next_in_queue = nullptr;
-      auto unmasked_back = unmask(back);
-      unmasked_back->next_in_queue.store(node, std::memory_order_release);
-      back = node;
+      std::atomic_thread_fence(std::memory_order_release);
+      auto unmasked_back = unmask(back.exchange(node, std::memory_order_relax));
+      unmasked_back->next_in_queue.store(node, std::memory_order_relax);
```

Comment:
This is not related to this PR, but should you also hold the Epoch so that unmasked_back cannot be deallocated before the write to the next_in_queue field?